prov/verbs/src/fi_verbs.c

```diff
@@ -2226,6 +2211,32 @@ static struct fi_ops_ep fi_ibv_msg_ep_base_ops = {
 	.tx_size_left = fi_no_tx_size_left,
 };
 
+static struct fi_ibv_msg_ep *fi_ibv_alloc_msg_ep()
+{
+	struct fi_ibv_msg_ep *ep;
+
+	ep = calloc(1, sizeof *ep);
+	if (!ep)
+		return NULL;
+
+	ep->tx_attr = calloc(1, sizeof *(ep->tx_attr));
+	if (!ep->tx_attr)
+		return NULL;
+
+	ep->rx_attr = calloc(1, sizeof *(ep->rx_attr));
+	if (!ep->rx_attr)
+		return NULL;
+
+	return ep;
+}
+
+static void fi_ibv_free_msg_ep(struct fi_ibv_msg_ep *ep)
+{
+	free(ep->tx_attr);
+	free(ep->rx_attr);
+	free(ep);
+}
```

Comment:
I'm merging this, but for a possible future change:  The way this is used, tx_attr and rx_attr should just be embedded structures, rather than pointers.